### PR TITLE
Do the TODO

### PR DESCRIPTION
**Pull request by TheDoubleK**
À la Betty Boo:
https://www.youtube.com/watch?v=-3U7FyKdBS4

### CODE REVIEW

File: app/main/views.py [hunk 1]
```diff
@@ -10,8 +10,10 @@ def index():
 
 @main.route('/service/<service_id>')
 def get_service_by_id(service_id):
-    # TODO: Return 404(or other appropriate response) for IDs that aren't found 
-    service = models.get_service(service_id)
+    try:
+        service = models.get_service(service_id)
+    except KeyError:
+        return Response("Service ID '%s' can not be found" % service_id, 404)
```
Comment (robyoung) [inline app/main/views.py]: For consistency with the data API could this use the `abort` function?

File: app/main/views.py [hunk 2]
```diff
@@ -10,8 +10,10 @@ def index():
 
 @main.route('/service/<service_id>')
 def get_service_by_id(service_id):
-    # TODO: Return 404(or other appropriate response) for IDs that aren't found 
-    service = models.get_service(service_id)
+    try:
+        service = models.get_service(service_id)
```
Comment (robyoung) [inline app/main/views.py]: I wonder if we could use the `first_or_404` pattern from SQLAlchemy and push this back up into the model layer. I'm not particularly opinionated either way, just raising the question.

File: app/main/views.py (hunk 1 above)
Comment (TheDoubleK) [inline app/main/views.py]: We could, but the abort doesn't seem to let us pass the message through - it just renders the default 404 handler.  So maybe it's better here to have the custom 404 response including the invalid service ID.

File: app/main/views.py (hunk 2 above)
Comment (TheDoubleK) [inline app/main/views.py]: We could... though I'm not sure we should develop the models here too much, as will all be replaced by search API once it exists.

**Pull request closed without merging**